quinn-udp/src/windows.rs

```diff
@@ -38,68 +61,180 @@ impl UdpSocketState {
+            tracing::error!("network stack does not support WSARecvMsg function");
+
+            return Err(io::Error::from(io::ErrorKind::Unsupported));
+        }
 
         if is_ipv4 {
-            let rc = unsafe {
-                WinSock::setsockopt(
-                    socket.0.as_raw_socket() as _,
-                    WinSock::IPPROTO_IP as _,
-                    WinSock::IP_DONTFRAGMENT as _,
-                    &sock_true as *const _ as _,
-                    mem::size_of_val(&sock_true) as _,
-                )
-            };
-            if rc == -1 {
-                return Err(io::Error::last_os_error());
-            }
+            set_socket_option(
+                &*socket.0,
+                WinSock::IPPROTO_IP,
+                WinSock::IP_DONTFRAGMENT,
+                OPTION_ON,
+            )?;
+
+            set_socket_option(
+                &*socket.0,
+                WinSock::IPPROTO_IP,
+                WinSock::IP_PKTINFO,
+                OPTION_ON,
+            )?;
+            set_socket_option(&*socket.0, WinSock::IPPROTO_IP, WinSock::IP_ECN, OPTION_ON)?;
         }
 
         if is_ipv6 {
-            let rc = unsafe {
-                WinSock::setsockopt(
-                    socket.0.as_raw_socket() as _,
-                    WinSock::IPPROTO_IPV6 as _,
-                    WinSock::IPV6_DONTFRAG as _,
-                    &sock_true as *const _ as _,
-                    mem::size_of_val(&sock_true) as _,
-                )
-            };
-            if rc == -1 {
-                return Err(io::Error::last_os_error());
-            }
+            set_socket_option(
+                &*socket.0,
+                WinSock::IPPROTO_IPV6,
+                WinSock::IPV6_DONTFRAG,
+                OPTION_ON,
+            )?;
+
+            set_socket_option(
+                &*socket.0,
+                WinSock::IPPROTO_IPV6,
+                WinSock::IPV6_PKTINFO,
+                OPTION_ON,
+            )?;
+
+            set_socket_option(
+                &*socket.0,
+                WinSock::IPPROTO_IPV6,
+                WinSock::IPV6_ECN,
+                OPTION_ON,
+            )?;
         }
 
+        // Opportunistically try to enable GRO
+        _ = set_socket_option(
+            &*socket.0,
+            WinSock::IPPROTO_UDP,
+            WinSock::UDP_RECV_MAX_COALESCED_SIZE,
+            // u32 per
+            // https://learn.microsoft.com/en-us/windows/win32/winsock/ipproto-udp-socket-options.
+            // Choice of 2^16 - 1 inspired by msquic.
+            u16::MAX as u32,
+        );
+
         let now = Instant::now();
         Ok(Self {
             last_send_error: Mutex::new(now.checked_sub(2 * IO_ERROR_LOG_INTERVAL).unwrap_or(now)),
         })
     }
 
-    pub fn send(&self, socket: UdpSockRef<'_>, transmits: &[Transmit]) -> Result<usize, io::Error> {
+    pub fn send(&self, socket: UdpSockRef<'_>, transmits: &[Transmit]) -> io::Result<usize> {
```

Comment:
Can we just revert the spelling change here from `Result<usize, io::Error>` to `io::Result<usize>`? IMO the former is more obvious.

Reply:
Sure i will do it

Reply:
@djc `unix.rs` has `send()` use `Result<usize, io::Error>` and `recv()` use `io::Result<usize>`. Want me to make all (`unix.rs` and `windows.rs`) use `Result<usize, io::Error>`  or `io::Result<usize>` ?

Reply:
Using `io::Result` is always preferred, and style fixes to existing code in that vein would be absolutely welcome. Not a blocker for this PR though.